activerecord/lib/active_record/counter_cache.rb

```diff
@@ -42,7 +42,7 @@ def reset_counters(id, *counters, touch: nil)
             [id]
           end
         else
-          Array(id)
+          where(primary_key => id).pluck(primary_key)
```

Comment:
I don't think it makes sense to make an additional query here. You end up making all callers pay the cost of the query instead of normalizing the value for the specific cases you need it

Reply:
I'm not sure yet if I'm understanding your problem. `to_param` will always return a string, why are you using it here?